gstreamer-sdp/src/s_d_p_message.rs

```diff
@@ -588,27 +664,27 @@ impl SDPMessage {
         }
     }
 
-    pub fn parse_buffer(data: &[u8]) -> Result<Self, SDPResult> {
+    pub fn parse_buffer(data: &[u8]) -> Result<Self, ()> {
         assert_initialized_main_thread!();
         unsafe {
             let size = data.len() as u32;
-            let mut msg = glib_ffi::g_malloc(mem::size_of::<ffi::GstSDPMessage>());
-            let result = from_glib(ffi::gst_sdp_message_parse_buffer(data.to_glib_none().0, size, &mut msg));
+            let msg = glib_ffi::g_malloc0(mem::size_of::<ffi::GstSDPMessage>()) as *mut ffi::GstSDPMessage;
+            let result = from_glib(ffi::gst_sdp_message_parse_buffer(data.to_glib_none().0, size, msg));
             match result {
-                SDPResult::ok => Some(from_glib_full(msg)),
-                _ => Err(result),
+                SDPResult::Ok => Ok(from_glib_full(msg)),
+                _ => Err(()),
```

Comment:
On error you need to `glib_ffi::g_free` the pointer, everywhere where `g_malloc` is used

Reply:
That is true...